### PR TITLE
Adding an Api for the homepage posts #12

### PR DESCRIPTION
**Pull request by idantb101**
creating an API for the homepage posts

### CODE REVIEW

**Comment by NickP123** [comment]
Looks good to me

File: package.json [hunk 1]
```diff
@@ -2,9 +2,9 @@
   "name": "healthyminds",
   "version": "1.0.0",
   "description": "",
-  "main": "index.js",
+  "main": "src/app.js",
   "scripts": {
-    "start": "nodemon app.js"
+    "start": "nodemon src/app.js"
```
Comment (Karyum) [inline package.json]: start script should be the production script, so it shouldnt have nodemon it should use node instead, nodemon is for development

File: src/db/queries/postQueries.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
+const connect = require('../db_connections.js');
+
+const getHomePagePosts = (cb) => {
+  connect.query(
+    `(SELECT * FROM posts where type = 'story' order by time_stamp desc LIMIT 2)
+      UNION ALL
+      (SELECT * FROM posts where type = 'advise' order by time_stamp desc LIMIT 2)
```
Comment (Karyum) [inline src/db/queries/postQueries.js]: can you explain whats going on in this query?

File: src/db/queries/postQueries.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+      UNION ALL
+      (SELECT * FROM posts where type = 'question' order by time_stamp desc LIMIT 2)`,
+    (err, res) => {
+      cb(err, res.rows);
```
Comment (Karyum) [inline src/db/queries/postQueries.js]: handle the error

File: src/routes/postsRoutes.js [hunk 4]
```diff
@@ -0,0 +1,18 @@
+const express = require('express');
+const postQueries = require('../db/queries/postQueries');
+
+const routes = () => {
+  const postsRoute = express.Router();
+  postsRoute.route('/homePagePosts').get((req, res) => {
+    postQueries.getHomePagePosts((err, homePagePosts) => {
+      if (err) {
+        console.log(err);
```
Comment (Karyum) [inline src/routes/postsRoutes.js]: `return` here so your code wouldn't hang incase of an error

File: src/db/queries/postQueries.js (hunk 2 above)
Comment (idantb101) [reply src/db/queries/postQueries.js]: creating three selects at the same time, everyone is different.

File: src/db/queries/postQueries.js (hunk 3 above)
Comment (idantb101) [reply src/db/queries/postQueries.js]: the error is handled in the callback

File: src/db/queries/postQueries.js (hunk 2 above)
Comment (Karyum) [reply src/db/queries/postQueries.js]: as in im guessing you are getting three types of posts from the database

**Pull request merged**